### PR TITLE
[Entity] added entity models

### PR DESCRIPTION
**Pull request by annton-shakur**
## Added entity models for:
* Accommodation
* Booking
* Payment
* User
* Role

### CODE REVIEW

File: src/main/java/com/example/accommodationbookingapp/model/Accommodation.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Data
+@SQLDelete(sql = "UPDATE accomodations SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
+@Table(name = "accomodations")
+public class Accommodation {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @JdbcTypeCode(SqlTypes.VARCHAR)
+    private Type type;
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/accommodationbookingapp/model/Accommodation.java:28]: add column  @Column(nullable = false) for all enum fields

File: src/main/java/com/example/accommodationbookingapp/model/Booking.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private Long id;
+    @Column(name = "check_in_date", nullable = false)
+    private LocalDate checkInDate;
+    @Column(name = "check_out_date", nullable = false)
+    private LocalDate checkOutDate;
+    @ManyToOne
+    @JoinColumn(name = "accommodation_id", nullable = false)
+    private Accommodation accommodation;
+    @ManyToOne
+    @JoinColumn(name = "user_id", nullable = false)
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/accommodationbookingapp/model/Booking.java:35]: @ManyToOne(fetch = FetchType.LAZY)

**Review by TarasYurkovskyi — CHANGES_REQUESTED** [review]

**Review by vadymhrnk — APPROVED** [review]
Nice job! LGTM

File: src/main/java/com/example/accommodationbookingapp/model/User.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/accommodationbookingapp/model/User.java:77]: change for return !isDeleted

**Review by TarasYurkovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/accommodationbookingapp/model/User.java (hunk 3 above)
Comment (TarasYurkovskyi) [reply src/main/java/com/example/accommodationbookingapp/model/User.java:77]: for all

**Review by TarasYurkovskyi — APPROVED** [review]
👍

File: src/main/java/com/example/accommodationbookingapp/model/Payment.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@Entity
+@Data
+@Table(name = "payments")
+public class Payment {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @JdbcTypeCode(SqlTypes.VARCHAR)
+    private Status status;
```
Comment (DPotapov93) [inline src/main/java/com/example/accommodationbookingapp/model/Payment.java:25]: I am not sure, but I think we need here the annotation "nullable = false" ?

**Pull request closed without merging**